test/test_preserve_bundler_env.rb

```diff
@@ -27,9 +30,11 @@ def test_usr2_restart_preserves_bundler_environment
     @pid = @server.pid
     connection = connect
     initial_reply = read_body(connection)
-    assert_match("Gemfile.bundle_env_preservation_test", initial_reply)
+    refute_match(bundle_gemfile, initial_reply)
```

Comment:
Seems this is a full revert of the work done in #1893, yes? The goal of #1893 was to allow a custom Gemfile name for apps that don't use the default.

Reply:
Seems #2133 is conflicting with #1893, yes? I'm not sure what to do.

Reply:
Here is the situation as I understand it:

1. Prior to #1893, it was not possible to run your app using a non-default Gemfile. Setting `BUNDLE_GEMFILE=Gemfile.rails6` (for example, as a way to test an app using a different version of Rails), would work at first, but as soon as the app was reloaded/restarted, then the environment variable would be reset to default. #1893 fixed that, but also introduced a bug: #2133.
2. #2133 details the bug where child processes inherit the **full path** of the Gemfile of the master process. Instead of the child process seeing the value of `BUNDLE_GEMFILE` as simply "Gemfile" or "Gemfile.rails6", the child process gets the full path to the gemfile, after its path has been expanded. That's bad.

If there is a way to have both the value preservation of #1893, but without the full path sharing detailed in #2133, then we'd like to have both.

Whew! I hope that makes sense.